gapil/resolver/inference.go

```diff
@@ -72,13 +72,10 @@ func inferNumber(rv *resolver, in *ast.Number, infer semantic.Type) semantic.Exp
 }
 
 func inferUnknown(rv *resolver, lhs semantic.Node, rhs semantic.Node) {
-	// We cannot infer unknown if assigning to _.
-	// This is ok if the unknown is still inferred later.
-	if _, isIgnore := lhs.(*semantic.Ignore); isIgnore {
-		return
-	}
 	u := findUnknown(rv, rhs)
 	if u != nil && u.Inferred == nil {
+		// This might fail to infer the expression (i.e. keeps it at nil).
```

Comment:
What is the LHS expression type that isn't `semantic.Ignore` and cannot be inferred?
Doesn't this change mean we can finish resolving without actually inferring an unknown? That seems really bad.

Reply:
The problematic code was this:
+  handle := ?
+  ctx := CreateContext(EGLContexts[share_context])
+  EGLContexts[handle] = ctx
+  Context2EGLContext[ctx] = handle
+  return handle

We can always finish without inferring - if there is no expression involving the handle, this code path will not be executed at all.

Reply:
What about code like this:

```
sub void foo(EGLContext share_context) {
    handle := ?
    ctx := CreateContext(EGLContexts[share_context])
    EGLContexts[handle] = ctx
}
```

Reply:
Behaves same as 
`sub void foo(EGLContext share_context) {
    handle := ?
    foo := handle
}`
did before.

That is, we never checked if unknowns are assigned, and that continues to be the case.
I think the generated code just files to compile, so it is just unfriendly